### DISCUSSION TITLE
how to enable logging on client for debug some CIFS problems

### DISCUSSION THREAD

**Discussion by Madtoyit** [ltsp/ltsp#296] [Q&A]
We are mount some CIFS network shares on FSTAB startup clients with ltsp.conf such as:
[clients]
POST_INIT_MKDIR_SHARE1="mkdir -p /mnt/myshare01"
FSTAB_MOUNT_SHARE1="//192.168.1.78/SHARE ...."

but sometimes some libreoffice edit files become corrupt and no longer readable.. very worrying.
we wanted to see in the client logs what happens. The strange thing is that from the external server (not the clients) this never happens and all go well.

Is there a way to enable / var / log / syslog on client?

Tks.

**Comment by alkisg** [comment]
Hi, all logs are viewable with `journalctl` or `journalctl -f` to follow or `journalctl -u unit` to see a specific unit.

The /var/log/syslog etc files are *duplicate* logs, most of them do not exist in other distributions like Fedora etc, and I think Debian will soon drop them as well. We do not want to keep these duplicates in LTSP clients in order to save RAM.

**Comment by Madtoyit** [comment]
tks Alkisg.
After you tips I can launch journalctl on client with terminal and a su admin login, so i can see logs.
But in logs there are some logs that i can't see but i find when i see log on user loged directly on server.
the log that i need to see are for a libreoffice open or modify file on a remote share mounted with CIFS on fstab.

in server i can see this log when modify an .odt file on my mnt/share:
Jun 22 10:49:09 ltspdeploy kernel: [ 1192.034467] audit: type=1400 audit(1592815749.402:2823): apparmor="ALLOWED" operation="open" profile="libreoffice-soffice" name=2F6D6E742F61733430306F63736170632F4455494C494F2F41646F626520392052656164657220496E7374616C6C61746F72652F6D6F63616363696E2F2E7E6C6F636B2E427542552E6F647423 pid=2782 comm="soffice.bin" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Jun 22 10:49:09 ltspdeploy kernel: [ 1192.035654] audit: type=1400 audit(1592815749.406:2824): apparmor="ALLOWED" operation="unlink" profile="libreoffice-soffice" name=2F6D6E742F61733430306F63736170632F4455494C494F2F41646F626520392052656164657220496E7374616C6C61746F72652F6D6F63616363696E2F2E7E6C6F636B2E427542552E6F647423 pid=2782 comm="soffice.bin" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
Jun 22 10:49:09 ltspdeploy kernel: [ 1192.037502] audit: type=1400 audit(1592815749.406:2825): apparmor="ALLOWED" operation="open" profile="libreoffice-soffice" name=2F6D6E742F61733430306F63736170632F4455494C494F2F41646F626520392052656164657220496E7374616C6C61746F72652F6D6F63616363696E2F427542552E6F6474 pid=2782 comm="soffice.bin" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0

but in client, when i modify an .odt in mnt/share no logs are written.
Is there some disabled log audit in client or how to increase log details?
Tks in advance

**Comment by alkisg** [comment]
Apparmor is disabled by default in LTSP clients, use this line to enable it and view its messages in the journal:

```shell
[clients]
KEEP_SYSTEM_SERVICES="apparmor"
```

**Comment by Madtoyit** [comment]
Tks Alkisg,
but  the problem is further complicating!
if i enable apparmor by KEEP_SYSTEM_SERVICES="apparmor" in client the libreoffice suite (the one I need to investigate my problem) crash with a "ubuntu popup crash" and no program in the libreoffice suite can be opened.
In client apparmor log there are many error of libreoffice wtih final "Fatal exception: Signal 11".

I can suppose  there is a substantial difficulty for libreoffice to function correctly or completely in the client.
for the tests performed it seems that only libreoffice crashes if I enable apparmor.

I hope you can investigate and thank you for any further solution

**Comment by alkisg** [comment]
Hi Madtoyit,

I haven't seen the issues you mention with LibreOffice while using LTSP with SSHFS and NFS for more than 10 years, so imagine they're related to CIFS.

Since this issue isn't related to LTSP at all, and since we're not using CIFS in schools here, I won't be investigating this personally. Of course feel free to keep commenting on this issue since some other LTSP users might be using CIFS or might be able to chime in with their ideas / workarounds about it.

**Comment by Madtoyit** [comment]
Ok tks
but the problem of crash libreoffice if i unmask apparmor isn't related to cifs. LO crash in any case.

For cifs it work perfectly with NAS and many network device. we have problem only with a specific server.
the need of cifs is a workarounds because ubuntu does not allow you to open files (of libreoffice) on a remote network if they are called via links on desktop... giomount problem...any suggestions?.

Tks alkisg for all.
